api/v1alpha1/base_types.go

```diff
@@ -44,6 +44,10 @@ type SecretTemplate struct {
 	Labels      map[string]string `json:"labels,omitempty"`
 	Annotations map[string]string `json:"annotations,omitempty"`
 	Key         *string           `json:"key,omitempty"`
+	UsernameKey *string           `json:"usernameKey,omitempty"`
+	PasswordKey *string           `json:"passwordKey,omitempty"`
+	HostKey     *string           `json:"hostKey,omitempty"`
+	PortKey     *string           `json:"portKey,omitempty"`
```

Comment:
You're correct, we're missing a `DatabaseKey`, I haven't figured out yet what to do when that value is `nil`, tho. I was trying to understand what would be the default value when left undefined, but it seems like the DSN is just not including it: https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/client/client.go#L60-L62. Do you think it would make sense to skip this if `nil`?

Reply:
Yep, leave it nil in that case. It would be something like:
```golang
	if mdbOpts.Database != "" && conn.Spec.SecretTemplate.DatabaseKey  {
		secretOpts.Data[*conn.Spec.SecretTemplate.DatabaseKey ] = []byte(mdbOpts.Database )
	}
```